framework/include/utils/PetscSupport.h

```diff
@@ -177,6 +229,11 @@ void colorAdjacencyMatrix(PetscScalar * adjacency_matrix,
                           std::vector<unsigned int> & vertex_colors,
                           const char * coloring_algorithm);
 
+/**
+ * disable a particular petsc flag
```

Comment:
More particular language would be good. Does this remove a flag if it exists? Does it tell the `PetscOptions` object to not add the flag if someone attempts to add it later? Both?

Reply:
We use the disable language repeatedly in the following doxygen. I think if you give a clear definition of what "disable" means here, you do not have to go and add it to all the following doc-strings

Reply:
If "disable" just means "dontAdd", then let's change the function name to match those that you added later in the file

Reply:
Once a petsc option is already actually set, I don't like to mess with it. My approach is to modify the `PetscOptions` object before options are set. So I think it makes sense to change `disable` to `dontAdd`. What do you like more, `dontAdd` or `dontSet`?

Reply:
Probably `dontAdd`